core/src/main/java/org/arquillian/smart/testing/scm/spi/ChangeResolver.java

```diff
@@ -5,7 +5,7 @@
 
 public interface ChangeResolver extends AutoCloseable {
 
-    Collection<Change> diff();
+    Collection<Change> diff(String strategy);
```

Comment:
Here I would create two methods `diff()` and `diff(String strategy)` with the javadoc explaining the usage and why there are two methods - also cases when the exception is thrown. `diff()` will use `null` as the default strategy value

Reply:
I would use `""` as default value for `diff()`

Reply:
Why? Is there any specific reason for it?
From my point of view, there shouldn't be anything set. The string containing something (even empty string) can be already considered as a string containing a content. Which means that if you don't want to set anything (just void) then you should pass the `null` as the best representation of "nothing".

Reply:
just to avoid NPE in future

Reply:
Having the check https://github.com/arquillian/smart-testing/pull/221#discussion_r144232376 if the strategy is null, then you won't get the NPE...
The `GitChangeResolver.diff` is something that can be externally used, so it should properly check if the parameter is null or not, otherwise, you will get unexpected NPE. At least this is what I would expect and how it is usually checked/used. If I check (in case of public classes) whether the string is not provided then as the first thing I check if it is null or not.

Reply:
Anyway, I don't really insist on having `null` as the default value when the `diff()` method is used, but please, put the check (if it is null or not) there. To avoid NPE.